src/main/java/com/cms/claimmanagement/insuranceCompany/claim/service/ClaimService.java

```diff
@@ -25,53 +28,43 @@ public String generateClaimId(String policyNo) {
         DateFormat df = new SimpleDateFormat("yyyy");
         String fourDigitYear = df.format(Calendar.getInstance().getTime());
 
-        return "CL" + fourPolicyNumbers + fourDigitYear;
+        return claimPrefix + fourPolicyNumbers + fourDigitYear;
 
     }
 
-    public ClaimResponseData saveNewClaim(insuredClaimRequestDTO claimRequest) {
-        ClaimDetails claim = new ClaimDetails();
-        claim.setClaimId(generateClaimId(claimRequest.getPolicyNo()));
-
-        PolicyEntity policy = new PolicyEntity();
-        policy.setPolicyNo(claimRequest.getPolicyNo());
-
-        claim.setPolicy(policy);
-
+    public ClaimResponseDTO saveNewClaim(InsuredClaimRequestDTO claimRequest) {
 
-        claim.setDateOfAccident(claimRequest.getDateOfAccident());
-        claim.setEstimatedLoss(claimRequest.getEstimatedLoss());
+        ClaimDetails claim = modelMapper.map(claimRequest, ClaimDetails.class);
 
         ClaimDetails savedClaim = claimRepository.save(claim);
-        ClaimResponseData claimId = new ClaimResponseData();
-        claimId.setClaimId(savedClaim.getClaimId());
+        ClaimResponseDTO claimId = modelMapper.map(savedClaim, ClaimResponseDTO.class);
 
 
         return claimId;
     }
 
-    public List<ClaimResponseData> getAllOpenClaims() {
+    public List<ClaimResponseDTO> getAllOpenClaims() {
 
         return claimRepository
                 .findAll()
                 .stream()
                 .filter(ClaimDetails::isClaimStatus)
                 .map(ClaimDetails::getClaimId)
```

Comment:
What is this map used for?

Reply:
I mean, the .map(ClaimDetails::getClaimId)

Reply:
I think it was supposed to get the claimId from all open claims and set to a dto. but I think  I missed something along the way.